app/controllers/donations_controller.rb

```diff
@@ -0,0 +1,65 @@
+class DonationsController < ApplicationController
+  skip_before_action :authenticate_user!
```

Comment:
You won't have the `current_user` method if you skip authenticating the user. You'll need to find a different way to allow anonymous donations